main.cpp

```diff
@@ -1,7 +1,314 @@
+#include <bits/stdc++.h> 
+#include <math.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <chrono>
+#include <fstream>
 #include <iostream>
-using namespace std;
-int main()
-{
-    cout << "Hello, World!";
+#include <random> 
+#include <vector>
+
+
+class MachineBlock{    
+    int start;
+    int end;
+    int taskNumber;
+    int operationNumber;
+    std::string blockType;
```

Comment:
I would recommend to avoid defining types as strings, because there is a risk that someone might misspell the word and then it might be hard to debug.... Instead I would suggest **enum**

> enum BlockType { 
> maintenance
> };

Then the variable blockType would be defines as:
**BlockType blockType;**

And assigning value would change from:

> this->blockType = blockType;

to:

> this->blockType = maintenance;

Also comaprisons would change from:

> if (machineBlock.get_blockType() == "maintenance")

to:

> if (machineBlock.get_blockType() == maintenance)

This approach is better, because if you misspell **maintenance** then you would see error at the time of compilation.